bin/test-build-on-all-distros

```diff
@@ -41,21 +41,24 @@ fi
 
 WORK_DIR="$(mktemp -dt hhvm.XXXXXXXX)"
 
+if which gtar
+then
+  TAR=gtar # the executable of GNU Tar in Homebrew 
+else
+  TAR=tar
+fi
 (
   set -x
   cd "$SOURCE_DIR"
-  git checkout-index -a -f --prefix="$WORK_DIR/hhvm-$VERSION/"
+  git ls-files --recurse-submodules |
+  egrep --invert-match -e '^(third-party/boost/boost/libs/[^/]+/(doc|test)|third-party/fb-mysql/mysql-5.6/(boost|rocksdb|xtrabackup|rqg|mysql-test)|hphp/test/(slow|quick|zend|zend7)/|"hphp/test/slow/)' |
```

Comment:
Can you replace this with reading the HHVM_REPO/.gitmodules file instead? The list here varies by version

Reply:
`grep` here is handling submodules. Instead `grep` is used to prune the source tree, like what `prune-source-tree` did in the previous version of this script. I cannot use `prune-source-tree` because this PR does not copy files into a directory to prune.

Reply:
Ah, sorry, I missed that this was sub-paths - but these third-party paths are paths that *used* to be inside submodules, and they do not exist in the source trees versions of HHVM that have migrated from submodules to using CMake's `ExternalProject_Add` for these.

third-party/boost and third-party/fb-mysql were converted away from submodules before 4.128, so the paths you're looking for here will never exist. The files will be inside the build directory instead, when downloaded by cmake.

https://github.com/facebook/hhvm/blob/HHVM-4.128/.gitmodules
https://github.com/facebook/hhvm/tree/HHVM-4.128/third-party/fb-mysql
https://github.com/facebook/hhvm/tree/HHVM-4.128/third-party/boost


The hphp/test/ cleanup is still needed.